From 1235f15f96a22e3751d535ee874150f2c48af4e9 Mon Sep 17 00:00:00 2001
From: rfishback <rfishback@anduril.com>
Date: Fri, 12 Jun 2026 11:46:46 -0700
Subject: [PATCH] overlay: add IMX708 camera overlays for p3768

Add the four DT files needed to support the Sony IMX708 (Raspberry Pi
Camera Module 3) on the Jetson Orin Nano dev kit (P3768/P3767-0000)
through nvargus.

- tegra234-camera-rbpcv3-imx708.dtsi: shared graph fragment defining
  the VI/NVCSI port graph for two IMX708 sensors at I2C address 0x1a,
  one mode (4608x2592 @ 14 fps RGGB10) per sensor.
- tegra234-p3767-camera-p3768-imx708-A.dts: CAM0-only overlay
  (port-index 1, serial_b, CAM0_PWDN reset).
- tegra234-p3767-camera-p3768-imx708-C.dts: CAM1-only overlay
  (port-index 2, serial_c, CAM1_PWDN reset).
- tegra234-p3767-camera-p3768-imx708-dual.dts: both ports.
- overlay/Makefile: register the three .dts files for .dtbo build.

Modeled on the IMX219 set
(tegra234-p3767-camera-p3768-imx219-{A,C,dual}.dts +
tegra234-camera-rbpcv2-imx219.dtsi). Single-mode for now; binned and
binned-cropped modes follow in a separate patch (Stage 2).

Originally based on RidgeRun's JP6.0 patch:
https://github.com/RidgeRun/NVIDIA-Jetson-IMX708-RPIV3
---
 overlay/Makefile                              |   3 +
 overlay/tegra234-camera-rbpcv3-imx708.dtsi    | 263 ++++++++++++++++++
 .../tegra234-p3767-camera-p3768-imx708-A.dts  | 188 +++++++++++++
 .../tegra234-p3767-camera-p3768-imx708-C.dts  | 188 +++++++++++++
 ...egra234-p3767-camera-p3768-imx708-dual.dts |  62 +++++
 5 files changed, 704 insertions(+)
 create mode 100644 overlay/tegra234-camera-rbpcv3-imx708.dtsi
 create mode 100644 overlay/tegra234-p3767-camera-p3768-imx708-A.dts
 create mode 100644 overlay/tegra234-p3767-camera-p3768-imx708-C.dts
 create mode 100644 overlay/tegra234-p3767-camera-p3768-imx708-dual.dts

diff --git a/overlay/Makefile b/overlay/Makefile
index c5b0cad..5556b8a 100644
--- a/overlay/Makefile
+++ b/overlay/Makefile
@@ -66,6 +66,9 @@ dtbo-y += tegra234-p3767-camera-p3768-imx219-A.dtbo
 dtbo-y += tegra234-p3767-camera-p3768-imx219-imx477.dtbo
 dtbo-y += tegra234-p3767-camera-p3768-imx477-C.dtbo
 dtbo-y += tegra234-p3767-camera-p3768-imx477-A.dtbo
+dtbo-y += tegra234-p3767-camera-p3768-imx708-A.dtbo
+dtbo-y += tegra234-p3767-camera-p3768-imx708-C.dtbo
+dtbo-y += tegra234-p3767-camera-p3768-imx708-dual.dtbo
 
 ifneq ($(dtb-y),)
 dtb-y := $(addprefix $(makefile-path)/,$(dtb-y))
diff --git a/overlay/tegra234-camera-rbpcv3-imx708.dtsi b/overlay/tegra234-camera-rbpcv3-imx708.dtsi
new file mode 100644
index 0000000..4ec49c9
--- /dev/null
+++ b/overlay/tegra234-camera-rbpcv3-imx708.dtsi
@@ -0,0 +1,263 @@
+// SPDX-License-Identifier: GPL-2.0-only
+// SPDX-FileCopyrightText: Copyright (c) 2022-2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+// SPDX-FileCopyrightText: Copyright (c) 2023, RidgeRun. All rights reserved.
+
+#include <dt-bindings/clock/tegra234-clock.h>
+#include <dt-bindings/gpio/tegra234-gpio.h>
+
+/ {
+	fragment-camera@0 {
+		target-path = "/";
+		__overlay__ {
+			tegra-capture-vi {
+				num-channels = <2>;
+				ports {
+					#address-cells = <1>;
+					#size-cells = <0>;
+					vi_port0: port@0 {
+						reg = <0>;
+						rpicam3_imx708_vi_in0: endpoint {
+							port-index = <1>;
+							bus-width = <2>;
+							remote-endpoint = <&rpicam3_imx708_csi_out0>;
+						};
+					};
+					vi_port1: port@1 {
+						reg = <1>;
+						rpicam3_imx708_vi_in1: endpoint {
+							port-index = <2>;
+							bus-width = <2>;
+							remote-endpoint = <&rpicam3_imx708_csi_out1>;
+						};
+					};
+				};
+			};
+
+			bus@0 {
+				host1x@13e00000 {
+					nvcsi@15a00000 {
+						num-channels = <2>;
+						#address-cells = <1>;
+						#size-cells = <0>;
+						csi_chan0: channel@0 {
+							reg = <0>;
+							ports {
+								#address-cells = <1>;
+								#size-cells = <0>;
+								csi_chan0_port0: port@0 {
+									reg = <0>;
+									rpicam3_imx708_csi_in0: endpoint@0 {
+										port-index = <1>;
+										bus-width = <2>;
+										remote-endpoint = <&rpicam3_imx708_out0>;
+									};
+								};
+								csi_chan0_port1: port@1 {
+									reg = <1>;
+									rpicam3_imx708_csi_out0: endpoint@1 {
+										remote-endpoint = <&rpicam3_imx708_vi_in0>;
+									};
+								};
+							};
+						};
+						csi_chan1: channel@1 {
+							reg = <1>;
+							ports {
+								#address-cells = <1>;
+								#size-cells = <0>;
+								csi_chan1_port0: port@0 {
+									reg = <0>;
+									rpicam3_imx708_csi_in1: endpoint@2 {
+										port-index = <2>;
+										bus-width = <2>;
+										remote-endpoint = <&rpicam3_imx708_out1>;
+									};
+								};
+								csi_chan1_port1: port@1 {
+									reg = <1>;
+									rpicam3_imx708_csi_out1: endpoint@3 {
+										remote-endpoint = <&rpicam3_imx708_vi_in1>;
+									};
+								};
+							};
+						};
+					};
+				};
+				cam_i2cmux {
+					i2c_0:i2c@0 {
+						imx708_cam0: rpicam3_imx708_a@1a {
+							compatible = "sony,imx708";
+							/* I2C device address */
+							reg = <0x1a>;
+							/* V4L2 device node location */
+							devnode = "video0";
+							/* Physical dimensions of sensor */
+							physical_w = "7.564";
+							physical_h = "5.476";
+							sensor_model = "imx708";
+							use_sensor_mode_id = "true";
+
+							/* IMX708_MODE_4608x2592_14fps */
+							mode0 {
+								mclk_khz = "24000";
+								num_lanes = "2";
+								tegra_sinterface = "serial_b";
+								phy_mode = "DPHY";
+								discontinuous_clk = "yes";
+								dpcm_enable = "false";
+								cil_settletime = "0";
+								lane_polarity = "6";
+
+								active_w = "4608";
+								active_h = "2592";
+								mode_type = "bayer";
+								pixel_phase = "rggb";
+								csi_pixel_bit_depth = "10";
+								readout_orientation = "90";
+								line_length = "4808";
+								inherent_gain = "1";
+								pix_clk_hz = "297600000";
+
+								gain_factor = "16";
+								framerate_factor = "1000000";
+								exposure_factor = "1000000";
+								min_gain_val = "16";
+								max_gain_val = "256";
+								step_gain_val = "1";
+								default_gain = "16";
+								min_hdr_ratio = "1";
+								max_hdr_ratio = "1";
+								min_framerate = "2000000";
+								max_framerate = "14000000";
+								step_framerate = "1";
+								default_framerate = "14000000";
+								min_exp_time = "500";
+								max_exp_time = "65487";
+								step_exp_time = "1";
+								default_exp_time = "1600";
+								embedded_metadata_height = "4";
+							};
+
+							ports {
+								#address-cells = <1>;
+								#size-cells = <0>;
+
+								port@0 {
+									reg = <0>;
+									rpicam3_imx708_out0: endpoint {
+										port-index = <1>;
+										bus-width = <2>;
+										remote-endpoint = <&rpicam3_imx708_csi_in0>;
+									};
+								};
+							};
+						};
+					};
+					i2c_1: i2c@1 {
+						imx708_cam1: rpicam3_imx708_c@1a {
+							compatible = "sony,imx708";
+							/* I2C device address */
+							reg = <0x1a>;
+							/* V4L2 device node location */
+							devnode = "video0";
+							/* Physical dimensions of sensor */
+							physical_w = "7.564";
+							physical_h = "5.476";
+							sensor_model = "imx708";
+							use_sensor_mode_id = "true";
+
+							/* IMX708_MODE_4608x2592_14fps */
+							mode0 {
+								mclk_khz = "24000";
+								num_lanes = "2";
+								tegra_sinterface = "serial_c";
+								phy_mode = "DPHY";
+								discontinuous_clk = "yes";
+								dpcm_enable = "false";
+								cil_settletime = "0";
+								lane_polarity = "6";
+
+								active_w = "4608";
+								active_h = "2592";
+								mode_type = "bayer";
+								pixel_phase = "rggb";
+								csi_pixel_bit_depth = "10";
+								readout_orientation = "90";
+								line_length = "4808";
+								inherent_gain = "1";
+								pix_clk_hz = "297600000";
+
+								gain_factor = "16";
+								framerate_factor = "1000000";
+								exposure_factor = "1000000";
+								min_gain_val = "16";
+								max_gain_val = "256";
+								step_gain_val = "1";
+								default_gain = "16";
+								min_hdr_ratio = "1";
+								max_hdr_ratio = "1";
+								min_framerate = "2000000";
+								max_framerate = "14000000";
+								step_framerate = "1";
+								default_framerate = "14000000";
+								min_exp_time = "500";
+								max_exp_time = "65487";
+								step_exp_time = "1";
+								default_exp_time = "1600";
+								embedded_metadata_height = "4";
+							};
+
+							ports {
+								#address-cells = <1>;
+								#size-cells = <0>;
+
+								port@0 {
+									reg = <0>;
+									rpicam3_imx708_out1: endpoint {
+										port-index = <2>;
+										bus-width = <2>;
+										remote-endpoint = <&rpicam3_imx708_csi_in1>;
+									};
+								};
+							};
+						};
+					};
+				};
+				tegra-camera-platform {
+					compatible = "nvidia, tegra-camera-platform";
+					num_csi_lanes = <4>;
+					max_lane_speed = <1500000>;
+					min_bits_per_pixel = <10>;
+					vi_peak_byte_per_pixel = <2>;
+					vi_bw_margin_pct = <25>;
+					max_pixel_rate = <750000>;
+					isp_peak_byte_per_pixel = <5>;
+					isp_bw_margin_pct = <25>;
+
+					modules {
+						cam_module0: module0 {
+							badge = "jakku_front_RPI3";
+							position = "front";
+							orientation = "1";
+							cam_module0_drivernode0: drivernode0 {
+								pcl_id = "v4l2_sensor";
+								devname = "imx708 9-001a";
+								sysfs-device-tree = "/sys/firmware/devicetree/base/bus@0/cam_i2cmux/i2c@0/rpicam3_imx708_a@1a";
+							};
+						};
+						cam_module1: module1 {
+							badge = "jakku_rear_RPI3";
+							position = "rear";
+							orientation = "1";
+							cam_module1_drivernode0: drivernode0 {
+								pcl_id = "v4l2_sensor";
+								devname = "imx708 10-001a";
+								sysfs-device-tree = "/sys/firmware/devicetree/base/bus@0/cam_i2cmux/i2c@1/rpicam3_imx708_c@1a";
+							};
+						};
+					};
+				};
+			};
+		};
+	};
+};
diff --git a/overlay/tegra234-p3767-camera-p3768-imx708-A.dts b/overlay/tegra234-p3767-camera-p3768-imx708-A.dts
new file mode 100644
index 0000000..38c750f
--- /dev/null
+++ b/overlay/tegra234-p3767-camera-p3768-imx708-A.dts
@@ -0,0 +1,188 @@
+// SPDX-License-Identifier: GPL-2.0-only
+// SPDX-FileCopyrightText: Copyright (c) 2023-2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+// SPDX-FileCopyrightText: Copyright (c) 2023, RidgeRun. All rights reserved.
+
+/dts-v1/;
+/plugin/;
+
+#define CAM0_RST	TEGRA234_MAIN_GPIO(H, 3)
+#define CAM0_PWDN	TEGRA234_MAIN_GPIO(H, 6)
+#define CAM1_PWDN	TEGRA234_MAIN_GPIO(AC, 0)
+#define CAM_I2C_MUX	TEGRA234_AON_GPIO(CC, 3)
+
+#include <dt-bindings/tegra234-p3767-0000-common.h>
+
+/ {
+	overlay-name = "Camera IMX708-A";
+	jetson-header-name = "Jetson 22pin CSI Connector";
+	compatible = JETSON_COMPATIBLE_P3768;
+
+	/* IMX708 (RPi Cam Module 3) on CSI PORT A / cam0 */
+	fragment@0 {
+		target-path = "/";
+		__overlay__ {
+			tegra-capture-vi {
+				num-channels = <1>;
+				ports {
+					#address-cells = <1>;
+					#size-cells = <0>;
+					vi_port0: port@0 {
+						reg = <0>;
+						rpicam3_imx708_vi_in0: endpoint {
+							port-index = <1>;
+							bus-width = <2>;
+							remote-endpoint = <&rpicam3_imx708_csi_out0>;
+						};
+					};
+				};
+			};
+			tegra-camera-platform {
+				compatible = "nvidia, tegra-camera-platform";
+				modules {
+					cam_module0: module0 {
+						badge = "jakku_front_RPI3";
+						position = "front";
+						orientation = "1";
+						cam_module0_drivernode0: drivernode0 {
+							pcl_id = "v4l2_sensor";
+							sysfs-device-tree = "/sys/firmware/devicetree/base/bus@0/cam_i2cmux/i2c@0/rpicam3_imx708_a@1a";
+						};
+					};
+				};
+			};
+			bus@0 {
+				host1x@13e00000 {
+					nvcsi@15a00000 {
+						num-channels = <1>;
+						#address-cells = <1>;
+						#size-cells = <0>;
+						csi_chan0: channel@0 {
+							reg = <0>;
+							ports {
+								#address-cells = <1>;
+								#size-cells = <0>;
+								csi_chan0_port0: port@0 {
+									reg = <0>;
+									rpicam3_imx708_csi_in0: endpoint@0 {
+										port-index = <1>;
+										bus-width = <2>;
+										remote-endpoint = <&rpicam3_imx708_out0>;
+									};
+								};
+								csi_chan0_port1: port@1 {
+									reg = <1>;
+									rpicam3_imx708_csi_out0: endpoint@1 {
+										remote-endpoint = <&rpicam3_imx708_vi_in0>;
+									};
+								};
+							};
+						};
+					};
+				};
+				cam_i2cmux {
+					status = "okay";
+					compatible = "i2c-mux-gpio";
+					#address-cells = <1>;
+					#size-cells = <0>;
+					mux-gpios = <&gpio_aon CAM_I2C_MUX GPIO_ACTIVE_HIGH>;
+					i2c-parent = <&cam_i2c>;
+					i2c@0 {
+						status = "okay";
+						reg = <0>;
+						#address-cells = <1>;
+						#size-cells = <0>;
+						rpicam3_imx708_a@1a {
+							reset-gpios = <&gpio CAM0_PWDN GPIO_ACTIVE_HIGH>;
+							compatible = "sony,imx708";
+							/* I2C device address */
+							reg = <0x1a>;
+							/* V4L2 device node location */
+							devnode = "video0";
+							/* Physical dimensions of sensor */
+							physical_w = "7.564";
+							physical_h = "5.476";
+							sensor_model = "imx708";
+							use_sensor_mode_id = "true";
+
+							/* IMX708_MODE_4608x2592_14fps */
+							mode0 {
+								mclk_khz = "24000";
+								num_lanes = "2";
+								tegra_sinterface = "serial_b";
+								phy_mode = "DPHY";
+								discontinuous_clk = "yes";
+								dpcm_enable = "false";
+								cil_settletime = "0";
+								lane_polarity = "6";
+
+								active_w = "4608";
+								active_h = "2592";
+								mode_type = "bayer";
+								pixel_phase = "rggb";
+								csi_pixel_bit_depth = "10";
+								readout_orientation = "90";
+								line_length = "4808";
+								inherent_gain = "1";
+								pix_clk_hz = "297600000";
+
+								gain_factor = "16";
+								framerate_factor = "1000000";
+								exposure_factor = "1000000";
+								min_gain_val = "16";
+								max_gain_val = "256";
+								step_gain_val = "1";
+								default_gain = "16";
+								min_hdr_ratio = "1";
+								max_hdr_ratio = "1";
+								min_framerate = "2000000";
+								max_framerate = "14000000";
+								step_framerate = "1";
+								default_framerate = "14000000";
+								min_exp_time = "500";
+								max_exp_time = "65487";
+								step_exp_time = "1";
+								default_exp_time = "1600";
+								embedded_metadata_height = "4";
+							};
+
+							ports {
+								#address-cells = <1>;
+								#size-cells = <0>;
+								port@0 {
+									reg = <0>;
+									rpicam3_imx708_out0: endpoint {
+										status = "okay";
+										port-index = <1>;
+										bus-width = <2>;
+										remote-endpoint = <&rpicam3_imx708_csi_in0>;
+									};
+								};
+							};
+						};
+					};
+					i2c@1 {
+						rpicam3_imx708_c@1a {
+							status = "disabled";
+						};
+					};
+				};
+				gpio@2200000 {
+					camera-control-output-low {
+						gpio-hog;
+						output-low;
+						gpios = <CAM0_RST 0>;
+						label = "cam0-rst";
+					};
+				};
+				gpio@6000d000 {
+					camera-control-output-low {
+						gpio-hog;
+						output-low;
+						gpios = < CAM1_PWDN 0  CAM0_PWDN 0>;
+						label = "cam1-pwdn", "cam0-pwdn";
+					};
+				};
+			};
+		};
+	};
+};
diff --git a/overlay/tegra234-p3767-camera-p3768-imx708-C.dts b/overlay/tegra234-p3767-camera-p3768-imx708-C.dts
new file mode 100644
index 0000000..196c610
--- /dev/null
+++ b/overlay/tegra234-p3767-camera-p3768-imx708-C.dts
@@ -0,0 +1,188 @@
+// SPDX-License-Identifier: GPL-2.0-only
+// SPDX-FileCopyrightText: Copyright (c) 2023-2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+// SPDX-FileCopyrightText: Copyright (c) 2023, RidgeRun. All rights reserved.
+
+/dts-v1/;
+/plugin/;
+
+#define CAM0_RST	TEGRA234_MAIN_GPIO(H, 3)
+#define CAM0_PWDN	TEGRA234_MAIN_GPIO(H, 6)
+#define CAM1_PWDN	TEGRA234_MAIN_GPIO(AC, 0)
+#define CAM_I2C_MUX	TEGRA234_AON_GPIO(CC, 3)
+
+#include <dt-bindings/tegra234-p3767-0000-common.h>
+
+/ {
+	overlay-name = "Camera IMX708-C";
+	jetson-header-name = "Jetson 22pin CSI Connector";
+	compatible = JETSON_COMPATIBLE_P3768;
+
+	/* IMX708 (RPi Cam Module 3) on CSI PORT C / cam1 */
+	fragment@0 {
+		target-path = "/";
+		__overlay__ {
+			tegra-capture-vi {
+				num-channels = <1>;
+				ports {
+					#address-cells = <1>;
+					#size-cells = <0>;
+					vi_port1: port@1 {
+						reg = <0>;
+						rpicam3_imx708_vi_in1: endpoint {
+							port-index = <2>;
+							bus-width = <2>;
+							remote-endpoint = <&rpicam3_imx708_csi_out1>;
+						};
+					};
+				};
+			};
+			tegra-camera-platform {
+				compatible = "nvidia, tegra-camera-platform";
+				modules {
+					cam_module1: module1 {
+						badge = "jakku_rear_RPI3";
+						position = "rear";
+						orientation = "1";
+						cam_module1_drivernode0: drivernode0 {
+							pcl_id = "v4l2_sensor";
+							sysfs-device-tree = "/sys/firmware/devicetree/base/bus@0/cam_i2cmux/i2c@1/rpicam3_imx708_c@1a";
+						};
+					};
+				};
+			};
+			bus@0 {
+				host1x@13e00000 {
+					nvcsi@15a00000 {
+						num-channels = <1>;
+						#address-cells = <1>;
+						#size-cells = <0>;
+						csi_chan1: channel@1 {
+							reg = <0>;
+							ports {
+								#address-cells = <1>;
+								#size-cells = <0>;
+								csi_chan1_port0: port@0 {
+									reg = <0>;
+									rpicam3_imx708_csi_in1: endpoint@2 {
+										port-index = <2>;
+										bus-width = <2>;
+										remote-endpoint = <&rpicam3_imx708_out1>;
+									};
+								};
+								csi_chan1_port1: port@1 {
+									reg = <1>;
+									rpicam3_imx708_csi_out1: endpoint@3 {
+										remote-endpoint = <&rpicam3_imx708_vi_in1>;
+									};
+								};
+							};
+						};
+					};
+				};
+				cam_i2cmux {
+					status = "okay";
+					compatible = "i2c-mux-gpio";
+					#address-cells = <1>;
+					#size-cells = <0>;
+					mux-gpios = <&gpio_aon CAM_I2C_MUX GPIO_ACTIVE_HIGH>;
+					i2c-parent = <&cam_i2c>;
+					i2c@0 {
+						rpicam3_imx708_a@1a {
+							status = "disabled";
+						};
+					};
+					i2c@1 {
+						status = "okay";
+						reg = <1>;
+						#address-cells = <1>;
+						#size-cells = <0>;
+						rpicam3_imx708_c@1a {
+							reset-gpios = <&gpio CAM1_PWDN GPIO_ACTIVE_HIGH>;
+							compatible = "sony,imx708";
+							/* I2C device address */
+							reg = <0x1a>;
+							/* V4L2 device node location */
+							devnode = "video0";
+							/* Physical dimensions of sensor */
+							physical_w = "7.564";
+							physical_h = "5.476";
+							sensor_model = "imx708";
+							use_sensor_mode_id = "true";
+
+							/* IMX708_MODE_4608x2592_14fps */
+							mode0 {
+								mclk_khz = "24000";
+								num_lanes = "2";
+								tegra_sinterface = "serial_c";
+								phy_mode = "DPHY";
+								discontinuous_clk = "yes";
+								dpcm_enable = "false";
+								cil_settletime = "0";
+								lane_polarity = "6";
+
+								active_w = "4608";
+								active_h = "2592";
+								mode_type = "bayer";
+								pixel_phase = "rggb";
+								csi_pixel_bit_depth = "10";
+								readout_orientation = "90";
+								line_length = "4808";
+								inherent_gain = "1";
+								pix_clk_hz = "297600000";
+
+								gain_factor = "16";
+								framerate_factor = "1000000";
+								exposure_factor = "1000000";
+								min_gain_val = "16";
+								max_gain_val = "256";
+								step_gain_val = "1";
+								default_gain = "16";
+								min_hdr_ratio = "1";
+								max_hdr_ratio = "1";
+								min_framerate = "2000000";
+								max_framerate = "14000000";
+								step_framerate = "1";
+								default_framerate = "14000000";
+								min_exp_time = "500";
+								max_exp_time = "65487";
+								step_exp_time = "1";
+								default_exp_time = "1600";
+								embedded_metadata_height = "4";
+							};
+
+							ports {
+								#address-cells = <1>;
+								#size-cells = <0>;
+								port@0 {
+									reg = <0>;
+									rpicam3_imx708_out1: endpoint {
+										status = "okay";
+										port-index = <2>;
+										bus-width = <2>;
+										remote-endpoint = <&rpicam3_imx708_csi_in1>;
+									};
+								};
+							};
+						};
+					};
+				};
+				gpio@2200000 {
+					camera-control-output-low {
+						gpio-hog;
+						output-low;
+						gpios = <CAM0_RST 0>;
+						label = "cam0-rst";
+					};
+				};
+				gpio@6000d000 {
+					camera-control-output-low {
+						gpio-hog;
+						output-low;
+						gpios = < CAM1_PWDN 0  CAM0_PWDN 0>;
+						label = "cam1-pwdn", "cam0-pwdn";
+					};
+				};
+			};
+		};
+	};
+};
diff --git a/overlay/tegra234-p3767-camera-p3768-imx708-dual.dts b/overlay/tegra234-p3767-camera-p3768-imx708-dual.dts
new file mode 100644
index 0000000..7f224e5
--- /dev/null
+++ b/overlay/tegra234-p3767-camera-p3768-imx708-dual.dts
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: GPL-2.0-only
+// SPDX-FileCopyrightText: Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+// SPDX-FileCopyrightText: Copyright (c) 2023, RidgeRun. All rights reserved.
+
+/dts-v1/;
+/plugin/;
+
+#define CAM0_RST	TEGRA234_MAIN_GPIO(H, 3)
+#define CAM0_PWDN	TEGRA234_MAIN_GPIO(H, 6)
+#define CAM1_PWDN	TEGRA234_MAIN_GPIO(AC, 0)
+#define CAM_I2C_MUX	TEGRA234_AON_GPIO(CC, 3)
+
+#include <dt-bindings/tegra234-p3767-0000-common.h>
+#include "tegra234-camera-rbpcv3-imx708.dtsi"
+
+/ {
+	overlay-name = "Camera IMX708 Dual";
+	jetson-header-name = "Jetson 22pin CSI Connector";
+	compatible = JETSON_COMPATIBLE_P3768;
+
+	fragment-camera-imx708@0 {
+		target-path = "/";
+		__overlay__ {
+			bus@0 {
+				cam_i2cmux {
+					status = "okay";
+					compatible = "i2c-mux-gpio";
+					#address-cells = <1>;
+					#size-cells = <0>;
+					i2c-parent = <&cam_i2c>;
+					mux-gpios = <&gpio_aon CAM_I2C_MUX GPIO_ACTIVE_HIGH>;
+					i2c@0 {
+						status = "okay";
+						reg = <0>;
+						#address-cells = <1>;
+						#size-cells = <0>;
+						rpicam3_imx708_a@1a {
+							reset-gpios = <&gpio CAM0_PWDN GPIO_ACTIVE_HIGH>;
+						};
+					};
+					i2c@1 {
+						status = "okay";
+						reg = <1>;
+						#address-cells = <1>;
+						#size-cells = <0>;
+						rpicam3_imx708_c@1a {
+							reset-gpios = <&gpio CAM1_PWDN GPIO_ACTIVE_HIGH>;
+						};
+					};
+				};
+				gpio@2200000 {
+					camera-control-output-low {
+						gpio-hog;
+						output-low;
+						gpios = <CAM0_RST 0>;
+						label = "cam0-rst";
+					};
+				};
+			};
+		};
+	};
+};
-- 
2.54.0

